static/app/views/dashboardsV2/orgDashboards.tsx

```diff
@@ -122,12 +122,26 @@ class OrgDashboards extends AsyncComponent<Props, State> {
   }
 
   renderBody() {
-    const {children} = this.props;
+    const {children, organization} = this.props;
     const {selectedDashboard, error} = this.state;
+    let dashboard = selectedDashboard;
+
+    if (organization.features.includes('dashboard-grid-layout')) {
+      // Ensure there are always tempIds for grid layout
+      // This is needed because there are cases where the dashboard
+      // renders before the onRequestSuccess setState is processed
+      // and will caused stacked widgets because of missing tempIds
+      dashboard = selectedDashboard
```

Comment:
Good call, I removed it and the behaviour was the same. I noticed a different bug with the context menu though. If you do any of the actions you get a toast that says "A dashboard with this name already exists", even though the frontend looks like a change has occurred, then clicking Edit Dashboard resets the dashboard to the state before doing any context menu actions. I checked and this is already on master, so different bug/PR?

Reply:
Thought about this a little bit. Actually not sure if we want to remove the setState entirely, since I think we still rely on it to update `selectedDashboard`, otherwise we might have `selectedDashboard` as null in some cases even though we did get a dashboard. I think we just want to remove the `assignTempId` map and keep the setState as `this.setState({[stateKey]: data})`

Reply:
nvm, spoke to nar, `onRequestSuccess` doesn't overwrite the success handler. The asynccomponent handleRequestSuccess sets the state key value by automatically